### PR TITLE
Add dem material relations

### PR DESCRIPTION
**Pull request by nathanfranklin**
This PR addresses #269 
- [x] add 3 new material relations (and 1 new CUBA attribute)
- [x] renamed CUBA attribute from  LN_OF_RESTITUTION_COEFFICIENT to  RESTITUTION_COEFFICIENT
- [x]  add info about the domain of these material relations

Note these changes require that H5_FILE_VERSION be incremented in whatever release the occur.

### CODE REVIEW

**Comment by nathanfranklin** [comment]
This is ready for review @simphony/fraunhofer-iwm  , @mehdisadeghi , @roigcarlo

**Comment by nathanfranklin** [comment]
@TobiasRasp , I used our meeting notes from December where we wrote up the descriptions.  Did I get everything correct?

File: simphony/core/material_relation_definitions.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   class_name : LennardJones
   allowed_number_materials: [1, 2]
   doc_description: Approximated potential energy of interaction between a pair of neutral atoms or molecules based on their distance of separation.
+  domain: [MD]
```
Comment (nathanfranklin) [inline simphony/core/material_relation_definitions.yml]: @ahashibon,  I added a "domain" so we could label the different material relations.  Is that okay? Is the name okay?  

Currently it is just used in the documentation (for example, an extra column is added to the a table)

Comment (ahashibon) [inline simphony/core/material_relation_definitions.yml]: That is fine, the material relation domain is unambiguous.

**Comment by mehdisadeghi** [comment]
It looks fine.

**Comment by nathanfranklin** [comment]
@TobiasRasp  + @ahashibon , I added the changes we discussed.

**Comment by nathanfranklin** [comment]
@TobiasRasp , is this okay to merge?

**Comment by TobiasRasp** [comment]
Yes, go ahead!

**Comment by nathanfranklin** [comment]
> Yes, I thikn a comment is appropriate here. I would also add an issue in github so that when time permits one could update the ci setup to use the latest version when/if it is fixed

@itziakos  , sounds good. I just added an issue and a comment.  Will merge once CI is through.

**Pull request merged**